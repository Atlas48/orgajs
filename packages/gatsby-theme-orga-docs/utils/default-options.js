const path = require('path')

const defaultOptions = {
  location: ['docs'],
  imageMaxWidth: 1380,
  components: {},
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
