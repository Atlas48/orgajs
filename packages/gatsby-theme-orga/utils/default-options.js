const path = require('path')

const defaultOptions = {
  contentPath: 'content',
  filter: () => true,
  pagination: 0,
  columns: 2,
  indexPath: '/',
  imageMaxWidth: 1380,
  categoryIndexPath: category => `/${category}`,
  tagIndexPath: tag => `/:${tag}:`,
  postPath: ({ category, export_file_name }) =>
    path.resolve('/', category || '', export_file_name || ''),
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
