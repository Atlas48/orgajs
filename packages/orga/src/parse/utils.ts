import { Node } from 'unist'
import { push } from '../node'
import { Lexer } from '../tokenize'
import { Parent, Token } from '../types'

export default (lexer: Lexer) => {
  const { peek, eat, save, restore } = lexer
  const collect = (stop: (n: Token) => boolean) => (container: Parent): Parent => {
    const token = peek()
    if (!token || stop(token)) return container
    eat()
    push(container)(token)
    return collect(stop)(container)
  }

  const skip = (predicate: (token: Token) => boolean): void => {
    const token = peek()
    if (token && predicate(token)) {
      eat()
      skip(predicate)
      return
    }
    return
  }

  const tryTo = <T extends Node>(
    parse: (lexer: Lexer) => T | undefined,
    action: (node: T) => void,
  ): boolean => {
    const savePoint = save()
    const node = parse(lexer)
    if (!node) {
      restore(savePoint)
      return false
    }
    action(node)
    return true
  }

  return {
    collect,
    skip,
    tryTo,
  }
}
