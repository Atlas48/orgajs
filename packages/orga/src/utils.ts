import { Token } from './types'
import { PhrasingContent } from 'orga';

const matchOperatorsRe = /[|\\{}()[\]^$+*?.]/g

export const escape = (str: string): string => {
  return str.replace(matchOperatorsRe, '\\$&')
}

export const isPhrasingContent = (token: Token): token is PhrasingContent => {
  return token.type.startsWith('text.')
    || token.type === 'footnote.reference'
    || token.type === 'link'
}
