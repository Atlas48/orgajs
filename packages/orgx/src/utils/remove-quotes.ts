export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')

export default removeQuotes
