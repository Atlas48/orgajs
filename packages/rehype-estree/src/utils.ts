export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')
