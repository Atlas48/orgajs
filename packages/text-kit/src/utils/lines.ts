import { Point, Position } from 'unist'
import { CoreAPI } from '../core'

export default <T extends CoreAPI>(core: T) => {
  const linePosition = (
    start: number | Point,
    end?: number | Point
  ): Position | undefined => {
    let result: Position | undefined

    if (typeof start === 'number') {
      result = linePosition(core.toPoint(start))
    } else {
      const s = core.bol(start.line)
      const e = core.eol(start.line)
      if (!s || !e) return undefined
      result = {
        start: s,
        end: e,
      }
    }

    if (end && result) {
      const endLR = linePosition(end)
      if (!endLR) return undefined
      result.end = endLR.end
    }

    return result
  }
  return {
    ...core,
    linePosition,
  }
}
