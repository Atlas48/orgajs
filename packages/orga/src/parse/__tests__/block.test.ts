import { parse } from './utils'

it('should parse normal block', () => {
  expect(
    parse(
      `
#+BEGIN_SRC javascript
console.log('hello')
console.log('world')
#+END_SRC
`.trim()
    )
  ).toMatchSnapshot()
})

it('should handle broken blocks (missing end)', () => {
  expect(
    parse(
      `
#+BEGIN_SRC javascript
console.log('hello')
console.log('world')
`.trim()
    )
  ).toMatchSnapshot()
})

it('should handle broken blocks (missing begin)', () => {
  expect(
    parse(
      `
console.log('hello')
console.log('world')
#+END_SRC
`.trim()
    )
  ).toMatchSnapshot()
})

it('can handle html export', () => {
  expect(
    parse(
      `
#+BEGIN_EXPORT html
<h1>hello</h1>
<p>world!</p>
#+END_EXPORT
`.trim()
    )
  ).toMatchSnapshot()

  expect(
    parse(
      `
#+HTML: <h1>Hello</h1>
`.trim()
    )
  ).toMatchSnapshot()
})

it('can handle affiliated keywords', () => {
  expect(
    parse(
      `
#+NAME: code_block
#+BEGIN_SRC javascript
console.log('hello world.')
#+END_SRC
`.trim()
    )
  ).toMatchSnapshot()
})

it('can handle invalid affiliated keywords', () => {
  expect(
    parse(
      `
#+NAME: code_block

#+BEGIN_SRC javascript
console.log('hello world.')
#+END_SRC
`.trim()
    )
  ).toMatchSnapshot()

  expect(
    parse(
      `
#+NOP: code_block
#+BEGIN_SRC javascript
console.log('hello world.')
#+END_SRC
`.trim()
    )
  ).toMatchSnapshot()
})
