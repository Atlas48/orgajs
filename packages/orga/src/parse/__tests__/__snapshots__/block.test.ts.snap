// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can handle affiliated keywords 1`] = `
Object {
  "children": Array [
    Object {
      "attributes": Object {
        "name": "code_block",
      },
      "name": "SRC",
      "params": Array [
        "javascript",
      ],
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 10,
          "line": 4,
          "offset": 79,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 19,
        },
      },
      "type": "block",
      "value": "
console.log('hello world.')",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 10,
      "line": 4,
      "offset": 79,
    },
    "start": Object {
      "column": 1,
      "line": 2,
      "offset": 19,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`can handle html export 1`] = `
Object {
  "children": Array [
    Object {
      "attributes": Object {},
      "name": "EXPORT",
      "params": Array [
        "html",
      ],
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 13,
          "line": 4,
          "offset": 61,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "block",
      "value": "
<h1>hello</h1>
<p>world!</p>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 13,
      "line": 4,
      "offset": 61,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`can handle html export 2`] = `
Object {
  "children": Array [
    Object {
      "parent": [Circular],
      "type": "html",
      "value": "<h1>Hello</h1>",
    },
  ],
  "properties": Object {},
  "type": "document",
}
`;

exports[`can handle invalid affiliated keywords 1`] = `
Object {
  "children": Array [
    Object {
      "attributes": Object {
        "name": "code_block",
      },
      "name": "SRC",
      "params": Array [
        "javascript",
      ],
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 10,
          "line": 5,
          "offset": 80,
        },
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 20,
        },
      },
      "type": "block",
      "value": "
console.log('hello world.')",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 10,
      "line": 5,
      "offset": 80,
    },
    "start": Object {
      "column": 1,
      "line": 3,
      "offset": 20,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`can handle invalid affiliated keywords 2`] = `
Object {
  "children": Array [
    Object {
      "attributes": Object {},
      "name": "SRC",
      "params": Array [
        "javascript",
      ],
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 10,
          "line": 4,
          "offset": 78,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 18,
        },
      },
      "type": "block",
      "value": "
console.log('hello world.')",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 10,
      "line": 4,
      "offset": 78,
    },
    "start": Object {
      "column": 1,
      "line": 2,
      "offset": 18,
    },
  },
  "properties": Object {
    "nop": "code_block",
  },
  "type": "document",
}
`;

exports[`should handle broken blocks (missing begin) 1`] = `
Object {
  "children": Array [
    Object {
      "attributes": Object {},
      "children": Array [
        Object {
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 1,
              "offset": 20,
            },
            "start": Object {
              "column": 1,
              "line": 1,
            },
          },
          "type": "text.plain",
          "value": "console.log('hello')",
        },
        Object {
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 2,
              "offset": 21,
            },
            "start": Object {
              "column": 21,
              "line": 1,
              "offset": 20,
            },
          },
          "type": "text.plain",
          "value": " ",
        },
        Object {
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 2,
              "offset": 41,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 21,
            },
          },
          "type": "text.plain",
          "value": "console.log('world')",
        },
      ],
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 1,
          "line": 3,
          "offset": 42,
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 3,
      "offset": 42,
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`should handle broken blocks (missing end) 1`] = `
Object {
  "children": Array [
    Object {
      "attributes": Object {},
      "children": Array [
        Object {
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 2,
              "offset": 43,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 23,
            },
          },
          "type": "text.plain",
          "value": "console.log('hello')",
        },
        Object {
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 3,
              "offset": 44,
            },
            "start": Object {
              "column": 21,
              "line": 2,
              "offset": 43,
            },
          },
          "type": "text.plain",
          "value": " ",
        },
        Object {
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 3,
              "offset": 64,
            },
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 44,
            },
          },
          "type": "text.plain",
          "value": "console.log('world')",
        },
      ],
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 21,
          "line": 3,
          "offset": 64,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 23,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 21,
      "line": 3,
      "offset": 64,
    },
    "start": Object {
      "column": 1,
      "line": 2,
      "offset": 23,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`should parse normal block 1`] = `
Object {
  "children": Array [
    Object {
      "attributes": Object {},
      "name": "SRC",
      "params": Array [
        "javascript",
      ],
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 10,
          "line": 4,
          "offset": 74,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "block",
      "value": "
console.log('hello')
console.log('world')",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 10,
      "line": 4,
      "offset": 74,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;
